README.md

```diff
@@ -12,13 +12,68 @@ Docker image `sourcegraph/lang-go` from Docker Hub.
 
 ### 🔐 Secure deployment 🔐
 
-Anyone that connects to the language server can access resources such as private
-code that the language server has access to.
+If you have private code, we recommend deploying the language server behind an
+auth proxy (such as HTTP basic authentication below), a firewall, or a VPN.
 
-We recommend deploying the language server behind an auth proxy or firewall and
-treating it like an authenticated user. Make sure you set `go.sourcegraphUrl` to
-the URL that the language server should use to reach Sourcegraph, which is
-likely different from the URL that end users use.
+### HTTP basic authentication
+
+You can prevent unauthorized access to the language server by enforcing HTTP basic authentication in nginx, which comes with the sourcegraph/server image. At a high level, you'll create a secret then put it in both the nginx config and in your Sourcegraph global settings so that logged-in users are authenticated when their browser makes requests to the Go language server.
+
+Here's how to set it up:
+
+Create an `.htpasswd` file in the Sourcegraph config directory with one entry:
+
+```
+$ htpasswd -c ~/.sourcegraph/config/.htpasswd langserveruser
+New password:
+Re-type new password:
+Adding password for user langserveruser
+```
+
+Add these to your Sourcegraph global settings:
+
+```
+  "go.serverUrl": "ws://langserveruser:PASSWORD@host.docker.internal:4389/go",
+  "go.sourcegraphUrl": "http://host.docker.internal:7080",
```

Comment:
given the config below I would of thought you would of used the same port for serverUrl as sourcegraphUrl.

Reply:
Oh you're right, I changed this to 7080 to match the `sourcegraphUrl`.